presto-parser/src/test/java/com/facebook/presto/sql/TestSqlTemplateFormatter.java

```diff
@@ -0,0 +1,158 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.Call;
+import com.facebook.presto.sql.tree.CallArgument;
+import com.facebook.presto.sql.tree.DecimalLiteral;
+import com.facebook.presto.sql.tree.DoubleLiteral;
+import com.facebook.presto.sql.tree.GenericLiteral;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.Statement;
+import com.facebook.presto.sql.tree.StringLiteral;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.sql.SqlFormatter.formatSqlAsTemplate;
+import static org.testng.Assert.assertEquals;
+
+public class TestSqlTemplateFormatter
+{
+    private static final SqlParser SQL_PARSER = new SqlParser();
+
+    @Test
+    public void testSqlFormatterBinaryLiteral()
+    {
+        String statement = "SELECT * FROM table1 WHERE binField = X''";
+        String templateStatement = "SELECT * FROM table1 WHERE binField = ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterBooleanLiteral()
+    {
+        String statement = "SELECT * FROM table1 WHERE a = true AND b = false";
+        String templateStatement = "SELECT * FROM table1 WHERE a = ? AND b = ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterCharLiteral()
+    {
+        assertTemplate("SELECT * WHERE c = CHAR 'abc'", "SELECT * WHERE c = ?");
+    }
+
+    @Test
+    public void testSqlFormatterDecimalLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new DecimalLiteral("5")),
+                new CallArgument("b", new DecimalLiteral("6"))));
+        String templateStatement = "CALL foo(a => ?, b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterDoubleLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new DoubleLiteral("5")),
+                new CallArgument("b", new DoubleLiteral("6"))));
+        String templateStatement = "CALL foo(a => ?, b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testRewriteGenericLiteral()
```

Comment:
Should this be named "testRewrite"?

Reply:
Updated as `testSqlTemplateFormatterGenericLiteral` to be consistent.